x-pack/plugins/observability/server/lib/rules/threshold/utils.ts

```diff
@@ -41,21 +43,18 @@ export const oneOfLiterals = (arrayOfLiterals: Readonly<string[]>) =>
       arrayOfLiterals.includes(value) ? undefined : `must be one of ${arrayOfLiterals.join(' | ')}`,
   });
 
-export const validateIsStringElasticsearchJSONFilter = (value: string) => {
+export const validateKQLStringFilter = (value: string) => {
   if (value === '') {
     // Allow clearing the filter.
     return;
   }
 
-  const errorMessage = 'filterQuery must be a valid Elasticsearch filter expressed in JSON';
   try {
-    const parsedValue = JSON.parse(value);
-    if (!isEmpty(parsedValue.bool)) {
-      return undefined;
-    }
-    return errorMessage;
+    kbnBuildEsQuery(undefined, [{ query: value, language: 'kuery' }], []);
```

Comment:
I asked about the differences between `fromKueryExpression` and `kbnBuildEsQuery`, here is the answer: ([link](https://elastic.slack.com/archives/CNKNE0PHN/p1690555665013879?thread_ts=1690533252.393779&cid=CNKNE0PHN))

> I don’t think there is a difference. buildEsQuery calls those methods under the hood
> It’s what you get when you call fromKueryExpression, and what you pass to toElasticsearchQuery